src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs

```diff
@@ -371,32 +371,62 @@ private static void SetAdapterLoggingSettings(IMessageLogger messageLogger, IRun
         var result = new Dictionary<LazyExtension<ITestDiscoverer, ITestDiscovererCapabilities>, IEnumerable<string>>();
         var sourcesForWhichNoDiscovererIsAvailable = new List<string>(sources);
 
+        sources = sources.Distinct().ToList();
```

Comment:
```suggestion
        sources = sources.Distinct();
```

Reply:
I think it's not needed IEnumerable should be accepted by `Where` etc...

Reply:
Ah the reason was to ensure that the two where queries below operate on a realized collection with distinct elements (so that the `Distinct` call does not need to be repeated when below queries operate on `sources`). Similar reason for the queries in the below variables too...

Reply:
ah ok so allocation vs computation...it's ok.